ui/src/app/components/runtime-configuration-panel.rtl.spec.tsx

```diff
@@ -1223,4 +1430,1154 @@ describe(RuntimeConfigurationPanel.name, () => {
+      screen.getByText(
+        /your environment currently has a reattachable disk, which will be unused after you apply this update\. /i
+      )
+    ).toBeTruthy();
+
+    togglePDRadioButton();
+
+    clickExpectedButton('Next');
+    clickExpectedButton('Update');
+    await waitFor(async () => {
+      runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+      expect(runtimeApiStub.runtime.gceWithPdConfig).not.toBeNull();
+    });
+    await waitFor(() => {}, { interval: 1000 });
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+      expect(runtimeApiStub.runtime.gceWithPdConfig).toBeUndefined();
+      // expect(runtimeApiStub.runtime.gceWithPdConfig).toBeNull();
+    });
+  });
+
+  it('should allow skipping disk deletion when detaching', async () => {
+    setCurrentRuntime(detachableDiskRuntime());
+    const disk = existingDisk();
+    setCurrentDisk(disk);
+
+    const { container } = await component();
+    await pickComputeType(container, ComputeType.Dataproc);
+
+    clickExpectedButton('Next');
+
+    expect(
+      screen.getByText(
+        /your environment currently has a reattachable disk, which will be unused after you apply this update\./i
+      )
+    ).toBeTruthy();
+
+    // Default option should be NOT to delete.
+    clickExpectedButton('Next');
+    clickExpectedButton('Update');
+    runtimeApiStub.runtime.status = RuntimeStatus.DELETED;
+
+    await waitFor(() => {
+      expect(runtimeApiStub.runtime.status).toEqual(RuntimeStatus.CREATING);
+      expect(disksApiStub.disk?.name).toEqual(disk.name);
+    });
+  });
+
+  it('should prevent runtime creation when running cost is too high for free tier', async () => {
+    setCurrentRuntime(null);
+    const { container } = await component();
+    clickExpectedButton('Customize');
+    const createButton = screen.getByRole('button', { name: 'Create' });
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(150);
+    // This should make the cost about $50 per hour.
+    await pickNumWorkers(200);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should prevent runtime creation when worker count is invalid', async () => {
+    setCurrentRuntime(null);
+    const { container } = await component();
+    clickExpectedButton('Customize');
+
+    await pickComputeType(container, ComputeType.Dataproc);
+    await pickStandardDiskSize(DATAPROC_MIN_DISK_SIZE_GB);
+    const createButton = screen.getByRole('button', { name: 'Create' });
+    await pickNumWorkers(0);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(1);
+    expectButtonElementDisabled(createButton);
+
+    await pickNumWorkers(2);
+    expectButtonElementEnabled(createButton);
+  });
+
+  it('should allow runtime creation when running cost is too high for user provided billing', async () => {
```

Comment:
this test seems redundant.  it's included in the next test.

Reply:
Yeah wasnt sure about it thought presence of billingAccountName: 'user provided billing' made a different. Removed